templates/modelExpt/run_modelComp_expt.m

```diff
@@ -0,0 +1,170 @@
+expt.timing.visualfbdur = 0.75; 
+
+%set up duration feedback parameters
+expt.durcalc.min_dur = .4; % set_exptDefaults normally sets to .25
+expt.durcalc.max_dur = .65;
+expt.durcalc.ons_thresh = 0.15;
+expt.durcalc.offs_thresh = 0.4;
+
+%% Set up duration practice
+exptDur = expt;
+exptDur.session = 'dur'; 
+exptDur.dataPath = fullfile(expt.dataPath,exptDur.session);
+if ~exist(exptDur.dataPath, 'dir')
+    mkdir(exptDur.dataPath)
+end
+exptDur.ntrials = 6;
+
+exptDur.shiftMags   = zeros(1,exptDur.ntrials);
+exptDur.shiftAngles = zeros(1,exptDur.ntrials);
+
+exptDur.words = {'head' 'dead' 'Ted'};
+exptDur.allWords = mod(0:exptDur.ntrials-1, numel(exptDur.words)) + 1;
+exptDur.listWords = exptDur.words(exptDur.allWords);
+
+exptDur.conds = {'noShift'};
+exptDur.allConds = ones(1, exptDur.ntrials);
+exptDur.listConds = exptDur.conds(exptDur.allConds);
+
+% set up duration feedback parameters
+exptDur.bDurFB = ones(1,exptDur.ntrials); % 1 x ntrials vector
+
+exptDur = set_exptDefaults(exptDur);
+
+exptfile = fullfile(exptDur.dataPath,'expt.mat');
+dur.expt = exptDur;
+save(exptfile, '-struct', 'dur')
+fprintf('Saved exptDur file: %s.\n',exptfile);
+
+%% Set up compensation experiment
+% stimuli and vowel list
+expt.words = {'head' 'dead' 'Ted'}; 
+nwords = length(expt.words);
+[vowels,~,ivowels] = unique(txt2arpabet(expt.words));
+if length(vowels) == length(ivowels), vowels = vowels(ivowels); end
+expt = set_missingField(expt,'vowels',vowels);
+
+expt.shiftDirs = {0 -1 1};
+expt.shiftNames = {'noShift' 'shiftDown' 'shiftUp'};
+expt.shiftMag = 125;
+expt.conds = {'noShift' 'shiftDown' 'shiftUp'};
+
+if bTestMode
+    expt.nblocks = 1;
+else
+    expt.nblocks = 6;
+end
+
+expt.ntrials_per_block = nwords*(2+3); %aka 15 trials per block.
+
+% Number of trials that don't receive perturbation
+if bTestMode 
+    nBaseline = 1;
+else 
+    nBaseline = 15;
+end
+
+% Pseudorandomize stimuli like this: In each 15-trial block, each word gets
+% an equal number of trials. Within a word, there are 2 perturbation
+% trials, and 3 unperturbed trials. Within a word's 2 perturbation trials,
+% there's 1 UP and 1 DOWN perturbation. No perturbation trials are adjacent, even across blocks.
+expt = randomize_stimuli(expt,1,3,nBaseline);
+
+% Set up ntrials, coherence, and direction
+expt.ntrials = expt.nblocks * expt.ntrials_per_block;
+
+% Set up breaks, expt.ntrials must be divisible by break frequency
+breakfrequency = expt.ntrials_per_block * 2;
+expt.breakTrials = breakfrequency:breakfrequency:expt.ntrials;  %pp breaks after baseline; then, every 2 blocks
+
+% set up duration feedback parameters
+expt.bDurFB = ones(1,expt.ntrials); %yes, give participant feedback about vowel duration
+
+% Set up shifts for the expt structure
+expt.allShiftDirs = expt.allConds; 
+expt.listShiftDirs = [expt.shiftDirs{expt.allConds}];
+expt.allShiftNames = expt.allShiftDirs;
+expt.listShiftNames = expt.shiftNames(expt.allShiftNames);
```

Comment:
I'm a bit confused here... shiftNames and conds are the same in this experiment, right? Can we just set allShiftNames = allConds and listShiftNames = listConds?